src/Tizen.Nlp/Tizen.Nlp/NlpProxy.cs

```diff
@@ -0,0 +1,271 @@
+/*
+* Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using Tizen.Applications;
+using Tizen.Applications.RPCPort;
+
+namespace Tizen.Nlp
+{
+    internal class Message : ProxyBase
+    {
+        public event EventHandler Connected;
+        public event EventHandler Disconnected;
+        public event EventHandler Rejected;
+
+        private bool _online;
+        private readonly string _appId = null;
+        private readonly Object _lock = new Object();
+        private readonly List<CallbackBase> _delegateList = new List<CallbackBase>();
+
+        public abstract class CallbackBase
+        {
+            internal int Id;
+            internal int SeqId;
+            internal bool Once;
+            private static volatile int _seqNum = 0;
+
+            public string Tag => Id.ToString() + "::" + SeqId.ToString();
+
+            protected CallbackBase(int delegateId, bool once)
+            {
+                Id = delegateId;
+                SeqId = _seqNum++;
+                Once = once;
+            }
+
+            internal virtual void OnReceivedEvent(Parcel p) { }
+
+            internal static void Serialize(Parcel h, CallbackBase param)
+            {
+                h.WriteInt(param.Id);
+                h.WriteInt(param.SeqId);
+                h.WriteBool(param.Once);
+            }
+
+            internal static void Deserialize(Parcel h, CallbackBase param)
+            {
+                param.Id = h.ReadInt();
+                param.SeqId = h.ReadInt();
+                param.Once = h.ReadBool();
+            }
+        }
+
+        public sealed class NotifyCb : CallbackBase
+        {
+            public NotifyCb(bool once = false) : base((int)DelegateId.NotifyCb, once)
+            {
+            }
+
+            public delegate void Callback(string sender, Bundle msg);
+            public event Callback Received;
```

Comment:
✓ DO use System.EventHandler<TEventArgs> instead of manually creating new delegates to be used as event handlers. 
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/event

Reply:
Hello Mr.Lee
sorry ,the xxxProxy.cs file was generated by TIDL command line ,
the link as below 
http://suprem.sec.samsung.net/confluence/pages/viewpage.action?spaceKey=APPF&title=TIDL

as previous commits , some  simple issue of rule ,e.g. variable  name , class name and uninitialized issue etc.  we can modify , but ,  this  callback or delegate calling method ,we are afraid to modify more deeply .  even resident effect on communication of TIDL  ,
 so ,i think we need to discuss with the TIDL team about this issue . could you  skip the issue like this area ?

thanks